pypeit/core/qa.py

```diff
@@ -339,6 +339,18 @@ def html_exp_pngs(exp_name, det):
     # Return
     return links, body
 
+
+def gen_qa_dir(qa_path):
+    """ Make the QA directory if it doesn't already exist
+
+    Args:
+        qa_path (str):
+            Path to the QA folder
+    """
+    if not os.path.exists(qa_path):
+        os.makedirs(qa_path)
+
+
 def gen_mf_html(pypeit_file, qa_path):
     """ Generate the HTML for a MasterFrame set
 
```

Comment:
The crash happens on line 384 (`with open(MF_filename,'w') as f:`) if the directory `qa_path` doesn't already exist.